src/full/Agda/Syntax/Scope/Base.hs

```diff
@@ -725,6 +725,16 @@ allNamesInScope' s =
   mergeNamesMany [ map (, nameSpaceAccess nsId) <$> inNameSpace ns
                  | (nsId, ns) <- scopeNameSpaces s ]
 
+-- | Look up a single name in the current scope.
+--
+-- This is equivalent to @Map.lookup n . allNamesInScope'@, but more efficient
+-- when only a single name needs to be looked up.
+findNameInScope :: InScope a => C.Name -> Scope -> [(a, Access)]
+findNameInScope n s =
+  [ (name, nameSpaceAccess nsId)
+  | (nsId, ns) <- scopeNameSpaces s
+  , name <- fromMaybe [] . Map.lookup n $ inNameSpace ns ]
```

Comment:
`Map.findWithDefault [] n` is probably epsilon-better than `fromMaybe [] . Map.lookup n`.